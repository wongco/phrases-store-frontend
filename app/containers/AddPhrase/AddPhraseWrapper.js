import styled from 'styled-components';

const AddPhraseWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`;

export default AddPhraseWrapper;
