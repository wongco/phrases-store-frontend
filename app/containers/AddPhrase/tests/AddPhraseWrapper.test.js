import React from 'react';
import { shallow, mount } from 'enzyme';
import { enzymeFind } from 'styled-components/test-utils';
import toJson from 'enzyme-to-json';
import AddPhraseWrapper from '../AddPhraseWrapper';

describe('<AddPhraseWrapper />', () => {
  it('renders without crashing', () => {
    shallow(<AddPhraseWrapper />);
  });

  it('matches snapshot for default criteria', () => {
    const wrapper = shallow(<AddPhraseWrapper />);
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });

  it('should render a prop', () => {
    const id = 'testId';
    const wrapper = mount(<AddPhraseWrapper id={id} />);
    const renderedComponent = enzymeFind(wrapper, AddPhraseWrapper);
    expect(renderedComponent.prop('id')).toBe('testId');
  });

  it('should render its text', () => {
    const children = 'Text';
    const wrapper = mount(<AddPhraseWrapper>{children}</AddPhraseWrapper>);
    const renderedComponent = enzymeFind(wrapper, AddPhraseWrapper);
    expect(renderedComponent.contains(children)).toBe(true);
  });
});
