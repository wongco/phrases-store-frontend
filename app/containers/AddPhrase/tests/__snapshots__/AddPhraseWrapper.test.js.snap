// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddPhraseWrapper /> matches snapshot for default criteria 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "AddPhraseWrapper-xlgpvl-0",
        "isStatic": true,
        "rules": Array [
          "display:flex;flex-direction:column;align-items:center;",
        ],
      },
      "displayName": "AddPhraseWrapper",
      "render": [Function],
      "styledComponentId": "AddPhraseWrapper-xlgpvl-0",
      "target": "div",
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
